axelor-human-resource/src/main/resources/views/Timesheet.xml

```diff
@@ -169,7 +169,7 @@
 	
 	<action-record name="action-timesheet-record-default" model="com.axelor.apps.hr.db.Timesheet">
 	    <field name="user" expr="eval:__user__"/>
-	    <field name="company" expr="eval: __user__.activeCompany"/>
+	    <field name="company" expr="eval: __user__.employee?.mainEmploymentContract?.payCompany ?: __user__.activeCompany"/>
 	</action-record>
```

Comment:
Should also applied on other hr form view (expense, leave request, ...)

Reply:
already done, see 6685984ccfa2bfad0398e0c06e71153d1a02c5ed